src/import/importConfiguration.ts

```diff
@@ -99,6 +100,22 @@ export function importConfiguration(yaml: YAMLConfiguration | string, file: stri
     );
     throw new Error('Minimal config not met');
   }
+  // There are some properties that are required to generate an IG
+  if (
+    !yaml.FSHOnly &&
+    MINIMAL_IG_ONLY_PROPERTIES.some(
+      (p: keyof YAMLConfiguration) =>
+        yaml[p] == null || (Array.isArray(yaml[p]) && (yaml[p] as any[]).length === 0)
+    )
+  ) {
+    logger.error(
+      'SUSHI minimally requires the following configuration properties to generate an IG: ' +
+        MINIMAL_IG_ONLY_PROPERTIES.join(', ') +
+        '.',
+      { file }
+    );
+    throw new Error('Minimal config not met');
+  }
```

Comment:
I imagine it might be annoying for someone if they are missing a core required property and get an error that `canonical` and `fhirVersion` are required, then they add those and get another error that `id` is required.  It would probably be a better user experience to list _everything_ required based on their config one time.

You could probably reduce some duplicate code as well if you did something more like:
```ts
const minimalProperties = yam.FSHOnly ? MINIMAL_CONFIG_PROPERTIES : [...MINIMAL_CONFIG_PROPERTIES, ...MINIMAL_IG_PROPERTIES];
// ...
```
Or if you define `MINIMAL_IG_PROPERTIES` to include the `MINIMAL_CONFIG_PROPERTIES`, then the above can be even simpler.

Reply:
That sounds like a reasonable change to me. Right now, the issue I'm still twisting around is the fact that both `copyrightYear` and `copyrightyear` are allowed as spellings of the same property, which means the current way of checking for minimal properties will have to accommodate that.

Reply:
Dang.  I guess you could make the minimal constants be lists of regular expressions instead of lists of strings... Then you can just include `/copyright[Yy]ear/`.  It seems a little silly but also seems like the least complex solution...